packages/api-server/src/serverManager.ts

```diff
@@ -82,13 +83,39 @@ export class ServerManager {
   }
 
   async restartApiServer() {
-    this.killApiServer()
+    await this.killApiServer()
     await this.startApiServer()
   }
 
-  killApiServer() {
-    this.httpServerProcess?.emit('exit')
-    this.httpServerProcess?.kill()
+  async killApiServer() {
+    if (!this.httpServerProcess) {
+      return
+    }
+
+    // Try to gracefully close the server
+    // If it doesn't close within 2 seconds, forcefully close it
+    await Promise.race([
+      new Promise<void>((resolve) => {
+        console.log(
+          chalk.yellow(
+            'API server did not exit within 2 seconds, forcefully closing it.',
+          ),
+        )
+        this.httpServerProcess!.on('exit', () => resolve())
+        this.httpServerProcess!.kill()
```

Comment:
**Nitpick**: Perhaps we should try to gracefully shut down the server first?

First by calling `server.close` and then `server.closeAllConnections`

https://nodejs.org/dist/latest-v18.x/docs/api/http.html#servercloseallconnections

Reply:
The variable name is misleading, we don't have `http` instance but a "ChildProcess" because of the way we init the server.

https://github.com/redwoodjs/redwood/blob/582ad46f778d1463f6ae70abf7a56ad2185dc460/packages/api-server/src/serverManager.ts#L69-L83

Reply:
Ah, ok! Let's resolve this then as it's not applicable.